pwnlib/tubes/process.py

```diff
@@ -276,8 +280,12 @@ def __init__(self, argv = None,
         #: Whether setuid is permitted
         self._setuid      = setuid if setuid is None else bool(setuid)
 
-        # Create the PTY if necessary
-        stdin, stdout, stderr, master, slave = self._handles(*handles)
+        if sys.platform.startswith("linux"):
+            # Create the PTY if necessary
+            stdin, stdout, stderr, master, slave = self._handles(*handles)
+        elif sys.platform.startswith("win"):
+            stdin, stdout = os.pipe()
+            stderr = stdout
```

Comment:
I would rather see:

```
if sys.platform.startswith("win"):
  # windows stuff
else:
  existing code
```

Separately, I think it is a good idea to NOT call `sys.platform.startswith(...)` every time.

I would recommend adding a value to `pwnlib.context.Context` which has a default value.  Perhaps it could be named `context.host_os`.

Another approach would be to add a `context.windows` property which is read-only / cached with a default value.  Then you just have to di:

```
if context.windows:
  # windows-specific
  if context.os == 'linux':
    # host is windows, target is linux
  elif context.os == 'windows':
    # host and target are windows
```

Reply:
I'd prefer §host_os§